<launch>
  <arg name="debugFlag" default="false"/>
  <arg name="gdbFlag" default="false"/>
  <node name="orb_slam2_rgbd" pkg="orb_slam2_ros" type="orb_slam2_ros_rgbd" output="screen" unless="$(eval gdbFlag)">

       <remap from="/camera/rgb/image_raw" to="/camera/color/image_rect_color" />
       <remap from="/camera/depth_registered/image_raw" to="/camera/depth/image_rect_raw" />

       <param name="publish_pointcloud" type="bool" value="true" />
       <param name="publish_pose" type="bool" value="true" />
       <param name="localize_only" type="bool" value="false" />
       <param name="reset_map" type="bool" value="false" />

       <!-- static parameters -->
       <param name="load_map" type="bool" value="false" />
       <param name="map_file" type="string" value="map.bin" />
       <param name="settings_file" type="string" value="$(find orb_slam2_ros)/orb_slam2/config/RealSenseD435RGBD_custom.yaml" />
       <param name="voc_file" type="string" value="$(find orb_slam2_ros)/orb_slam2/Vocabulary/ORBvoc.txt" />

       <param name="pointcloud_frame_id" type="string" value="map" />
       <param name="camera_frame_id" type="string" value="camera_link" />
       <param name="min_num_kf_in_map" type="int" value="5" />

       <!-- Debug flag to enable debug logging -->
       <param name="debugFlag" type="bool" value="$(arg debugFlag)"/>
   </node>

   <!-- Run with gdb if debug flag -->
   <node name="orb_slam2_rgbd" pkg="orb_slam2_ros" type="orb_slam2_ros_rgbd" output="screen" launch-prefix="valgrind" if="$(eval gdbFlag)">

     <remap from="/camera/rgb/image_raw" to="/camera/color/image_rect_color" />
     <remap from="/camera/depth_registered/image_raw" to="/camera/depth/image_rect_raw" />

     <param name="publish_pointcloud" type="bool" value="true" />
     <param name="publish_pose" type="bool" value="true" />
     <param name="localize_only" type="bool" value="false" />
     <param name="reset_map" type="bool" value="false" />

     <!-- static parameters -->
     <param name="load_map" type="bool" value="false" />
     <param name="map_file" type="string" value="map.bin" />
     <param name="settings_file" type="string" value="$(find orb_slam2_ros)/orb_slam2/config/RealSenseD435RGBD_custom.yaml" />
     <param name="voc_file" type="string" value="$(find orb_slam2_ros)/orb_slam2/Vocabulary/ORBvoc.txt" />

     <param name="pointcloud_frame_id" type="string" value="map" />
     <param name="camera_frame_id" type="string" value="camera_link" />
     <param name="min_num_kf_in_map" type="int" value="5" />

     <!-- Debug flag to enable debug logging -->
     <param name="debugFlag" type="bool" value="$(arg debugFlag)"/>
 </node>
</launch>
